### DISCUSSION TITLE
Optional anonymous users on GitHub disussions.

### DISCUSSION THREAD

**Discussion by SageofPHY6Paths** [community/community#7069] [Discussions]
Can something like GitHub Discussions Anonymous Bot, which by default is disabled but we can enable it for a repository, be implemented? Not needed for issues, only for discussions. It will help non GitHub users greatly.

For large projects anonymous users might spam so it will be disabled. But for small projects it will be very useful.

**Comment by VarunAgw** [comment]
Yeah, I need a platform like github discussion but for anonymous user who are not technical.

**Reply by wasiiff** [reply]
Yeah

**Comment by cungminh2710** [comment]
this would be great for non-tech users (ie. sales, marketing) to feedback and refine roadmap, which can be linked to internal issues by devs

**Comment by wasiiff** [comment]
Yeah i would also like to get familiar to it

**Comment by Shree1194** [comment]
Useful for small projects, but moderation and spam are probably the blockers.If implemented, it would need to be opt-in, disabled by default, and backed by strong rate-limiting or approval to avoid abuse.

**Comment by henock-lobu** [comment]
This is an interesting idea, especially for improving accessibility in smaller communities.
Allowing anonymous participation in Discussions could lower the barrier for users who are not familiar with GitHub or do not want to create an account just to ask a question. This could be particularly useful for projects that target non-technical audiences.
However, as you mentioned, moderation would be a key challenge. Even if the feature is disabled by default, enabling anonymous contributions could increase the risk of spam and low quality content. To address this, some safeguards could be considered, such as rate limiting, CAPTCHA verification, or requiring moderator approval before publishing messages.
A possible compromise would be to allow maintainers to enable anonymous posting with strict controls. For example, anonymous users could submit messages that are reviewed before becoming visible, or have limited interaction capabilities.
Overall, this feature could be valuable for small projects and communities, as long as proper moderation tools are included to maintain quality and prevent abuse.
